source/modulo_components/src/Component.cpp

```diff
@@ -60,4 +61,10 @@ bool Component::on_execute_callback() {
 std::shared_ptr<state_representation::ParameterInterface> Component::get_parameter(const std::string& name) const {
   return ComponentInterface::get_parameter(name);
 }
+
+void Component::raise_error() {
+  ComponentInterface::raise_error();
+  this->set_predicate("in_error_state", true);
+  this->cancel_step();
```

Comment:
I feel like we might as well call `finalize_interfaces()` here, and then do the cancel_step logic inside of that call. That way we could remove the `cancel_step()` method entirely